readthedocs/search/tasks.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from django.apps import apps
+from django_elasticsearch_dsl.registries import registry
+
+from readthedocs.worker import app
+
+log = logging.getLogger(__name__)
+
+
+def _get_index(indices, index_name):
+    for index in indices:
+        if str(index) == index_name:
+            return index
+
+
+def _get_document(model, document_class):
+    documents = registry.get_documents(models=[model])
+
+    for document in documents:
+        if str(document) == document_class:
+            return document
```

Comment:
What is the logic here for? Will there be multiple documents for a model ever? Also needs a docstring :)

Reply:
Yes. there can be multiple documents for a model

```python
class FooDocument(DocType):
    ...
    class Meta:
        model = Bar

class NewFooDocument(DocType):
    ...
    class Meta:
        model = Bar
```